pkg/volume/util/device_util_linux.go

```diff
@@ -202,10 +202,22 @@ func (handler *deviceHandler) GetISCSIPortalHostMapForTarget(targetIqn string) (
 				portal := strings.TrimSpace(string(addr)) + ":" +
 					strings.TrimSpace(string(port))
 				portalHostMap[portal] = hostNumber
+				// For IPv6 addresses add an entry with IP address enclosed in []
```

Comment:
There already helpers for that https://pkg.go.dev/net#SplitHostPort and JoinHostPort